rules-xml/api/src/main/java/org/jboss/windup/rules/apps/xml/condition/XmlFile.java

```diff
@@ -358,15 +365,22 @@ else if (iterated instanceof XmlFileModel)
                         continue;
                     }
                 }
-                if (publicId != null && !publicId.isEmpty())
+                if (( publicId != null && !publicId.isEmpty() ) || dtdNamespace!=null)
                 {
                     DoctypeMetaModel doctype = xml.getDoctype();
-                    if (doctype == null || doctype.getPublicId() == null
-                                || !doctype.getPublicId().matches(publicId))
+                    if (doctype == null )
                     {
                         continue;
                     }
-                    else if (xpathString == null)
+                    if(publicId != null && ( (doctype.getPublicId() == null) || !doctype.getPublicId().matches(publicId))) {
+                        continue;
+                    }
+                    if(dtdNamespace != null && ( (doctype.getSystemId() == null) || !doctype.getSystemId().matches(dtdNamespace))) {
+                        continue;
+                    }
```

Comment:
this is not very nice. I have created separate https://issues.jboss.org/browse/WINDUP-887 for refactoring XmlFile into multiple subconditions